### PR TITLE
Add on

### PR DESCRIPTION
**Pull request by maryamariyan**
I'm trying to make changes to ExternDll.Odbc32 to make it adhere to interop conventions here:
https://github.com/dotnet/corefx/blob/42ce0e696e08e190aceaf8613e47e67d57178550/Documentation/coding-guidelines/interop-guidelines.md

still work in progress. need to figure out why it's not working yet

### CODE REVIEW

File: src/Common/src/Interop/Linux/Interop.Libraries.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    private static partial class Libraries
+    {
+        internal const string Odbc32 = "libodbc.2.dylib";
```
Comment (safern) [inline src/Common/src/Interop/Linux/Interop.Libraries.cs]: You could instead add this in Interop/Unix/Interop.Libraries.cs I see that both OSX and this one are the exact same value.

File: src/Common/src/Interop/OSX/Interop.Libraries.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ private static partial class Libraries
         internal const string libproc = "libproc";
         internal const string LibSystemCommonCrypto = "/usr/lib/system/libcommonCrypto";
         internal const string LibSystemKernel = "/usr/lib/system/libsystem_kernel";
+        internal const string Odbc32 = "libodbc.2.dylib";
```
Comment (safern) [inline src/Common/src/Interop/OSX/Interop.Libraries.cs]: Once you add it into the Unix file you can delete this one.

File: src/System.Data.Odbc/src/System.Data.Odbc.csproj [hunk 3]
```diff
@@ -111,13 +111,19 @@
...
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetsLinux)' == 'true' And '$(IsPartialFacadeAssembly)' != 'true' ">
-    <Compile Include="Common\System\Data\Common\ExternDll.Linux.cs" />
+    <Compile Include="$(CommonPath)\Interop\Linux\Interop.Libraries.cs">
+      <Link>Common\Interop\Linux\Interop.Libraries.cs</Link>
+    </Compile>
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetsOsx)' == 'true' And '$(IsPartialFacadeAssembly)' != 'true' ">
-    <Compile Include="Common\System\Data\Common\ExternDll.Osx.cs" />
+    <Compile Include="$(CommonPath)\Interop\OSX\Interop.Libraries.cs">
```
Comment (safern) [inline src/System.Data.Odbc/src/System.Data.Odbc.csproj:119]: Once you unify the change into the Unix file, you can delete both ItemGroups the TargetsOsx and TargetsLinux ones and then only use one with TargetsUnix and include the file there, i.e: 

```xml
<ItemGroup Condition=" '$(TargetsUnix)' == 'true' ">
   <Compile Include="$(CommonPath)\Interop\Unix\Interop.Libraries.cs">
       <Link>Common\Interop\Unix\Interop.Libraries.cs</Link>
    </Compile>
</ItemGroup>
```

Also, `IsPartialFacadeAssembly` is only set to true when `TargetGroup` == `netfx` which can't be built on non windows environments, so it wouldn't be needed for this `ItemGroup` as a condition.

File: src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs [hunk 4]
```diff
@@ -72,6 +72,32 @@ public void ProcessStart_UseShellExecute_OnLinux_ThrowsIfNoProgramInstalled()
...
+            string programToOpen = s_allowedProgramsToRun.FirstOrDefault(program => IsProgramInstalled(program));
+            string fileToOpen, folderToOpen;
+            folderToOpen = Environment.CurrentDirectory;
+
+            fileToOpen = GetTestFilePath() + ".txt";
+            File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled)}");
+
+            Console.WriteLine($"({nameof(ProcessStart_UseShellExecute_DirectoryAndFileWithSameName_Success)}) fileToOpen: {fileToOpen} and folderToOpen: {folderToOpen}" );
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || programToOpen != null)
```
Comment (safern) [inline src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs]: `PlatformDetection.IsOSX` instead of RuntimeInformation...

**Review by safern — COMMENTED** [review]
Left some comments

File: src/Common/src/Interop/OSX/Interop.Libraries.cs (hunk 2 above)
Comment (maryamariyan) [reply src/Common/src/Interop/OSX/Interop.Libraries.cs]: My mistake. It's supposed to be different for linux and that's why I added the new file

**Pull request closed without merging**